proposals/0026/README.md

```diff
@@ -0,0 +1,275 @@
+Operators have numerous options to use and deploy Kubernetes, so in order to mitigate concerns around complexity, we will provide extensive documentation on different alternatives to meet an operator's objectives.
+
+Examples include:
+* A cloud provider managed Kubernetes solution (GKE, AKS, EKS)
+* A single host [K3s][k3s] cluster
+* A multi-host on-premise cluster
+
+<!-- Link found when Google searching "Kubernetes baggage" -->
+[k8s-baggage]: https://www.redbubble.com/shop/kubernetes+bags
+[k8s-af]: https://k8s.af/
+[kcp]: https://github.com/kcp-dev/kcp/
+[k3s]: https://k3s.io
+
+### User Experience
+
+In order to use Tinkerbell, clients would interact with the Kubernetes API.
+In order to provision a machine, the steps would be:
+
+1. User creates a Hardware CRD object in Kubernetes.
+   This is analogous to the current `tink hardware push < hardware.json` command.
+1. User creates a Template CRD object in Kubernetes.
+   This is analogous to the current `tink template create < template.json` command
+1. User creates a Workflow CRD object in Kubernetes.
+   All the user would need to specify in the object `.spec` would be a reference to the Template object, and mappings for devices.
+   This is analogous to the current `tink workflow create -t $TEMPLATE_ID -r "{\"device_1\":\"$MAC_ADDR\"}` command.
+1. The Tinkerbell API would include a Kubernetes workflow controller that would watch for created workflows, and fill out the `.status` with most of the logic that currently exists in the [`CreateWorkflow()`][createwfrpc] RPC.
+1. The Tinkerbell API could subscribe to Workflow CRD changes, and stream them on to Tinkerbell worker clients over the existing [`GetWorkflowContexts()`][getwfctxs] streaming RPC.
+1. Tinkerbell worker clients would continue to call the existing Tinkerbell APIs to execute workflows.
+  The Tinkerbell API would store updates in a Workflow CRD `.status`
+
+[createwfrpc]: https://github.com/tinkerbell/tink/blob/b217be8/grpc-server/workflow.go#L19-L72
+[getwfctxs]: https://github.com/tinkerbell/tink/blob/a56e5cf9/protos/workflow/workflow.proto#L75
+
+## System-context-diagram
+
+See above diagrams.
+
+## APIs
+
+The following [Tinkerbell Workflow APIs][wf-apis] are used by the Tink worker, and would remain.
+All other Tinkerbell APIs would be removed.
+
+[wf-apis]: https://github.com/tinkerbell/tink/blob/f6aa3930/protos/workflow/workflow.proto
+
+```protobuf
+service WorkflowService {
+  rpc GetWorkflowContexts(WorkflowContextRequest) returns (stream WorkflowContext) {}
+  rpc GetWorkflowActions(WorkflowActionsRequest) returns (WorkflowActionList) {}
+  rpc ReportActionStatus(WorkflowActionStatus) returns (Empty) {}
+  rpc GetWorkflowData(GetWorkflowDataRequest) returns (GetWorkflowDataResponse) {}
+  rpc UpdateWorkflowData(UpdateWorkflowDataRequest) returns (Empty) {}
+}
+```
+
+The structure of the Kubernetes CRD types will start out similar to those defined in the [Tinkerbell CAPI Provider][capt-types].
```

Comment:
It would be interesting to see in this proposal some description of the choices made in CAPT and how that could change.  Are the resources namespace or cluster scoped, for example, and why?

Reply:
I'll push those specifics into the actual PR reviews